### PR TITLE
cmake Xcode recommended build settings

### PR DESCRIPTION
**Pull request by Coeur**
Adopting the Xcode recommended build settings in cmake:
- Xcode "new build system" (POLICY CMP0114)
- ~`-fobjc-weak`~
- ~`-dead_strip`~

### CODE REVIEW

**Comment by ckerr** [comment]
@Coeur does this fix something that was broken, or does it improve the xcode build?

I'm trying to understand whether this is semver minor or patch

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,9 @@ cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
 if(POLICY CMP0092)
     cmake_policy(SET CMP0092 NEW)
 endif()
+if(POLICY CMP0114)
```
Comment (nevack) [inline CMakeLists.txt:6]: @mikedld JFYI https://cmake.org/cmake/help/latest/policy/CMP0114.html

**Comment by nevack** [comment]
> @Coeur does this fix something that was broken, or does it improve the xcode build?
> 
> I'm trying to understand whether this is semver minor or patch

I guess minor, but i don't see how this can break something.
But let's hear @Coeur of course

File: CMakeLists.txt (hunk 1 above)
Comment (mikedld) [reply CMakeLists.txt:6]: Not sure why you're tagging me, but also not sure how this relates to Xcode and its recommended settings...

**Comment by Coeur** [comment]
@ckerr building from the Xcode project and from cmake has various differences: this change reduces those differences, which helps future debugging or crash analysis. This PR is partly the equivalent of #3940, but for cmake.

File: macosx/CMakeLists.txt [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
+    $<$<COMPILE_LANGUAGE:C,CXX>:-fobjc-weak>
+
     # GNU extensions are good practice in Objective-C
-    -Wno-gnu)
+    $<$<COMPILE_LANGUAGE:C,CXX>:-Wno-gnu>
 
-find_program(ACTOOL_EXECUTABLE actool REQUIRED)
-find_program(CODESIGN_EXECUTABLE codesign REQUIRED)
+    # Our #warnings are good practice in IDE development, not in command-line compilation
+    $<$<AND:$<NOT:$<STREQUAL:CMAKE_GENERATOR,Xcode>>,$<COMPILE_LANGUAGE:C,CXX>>:-Wno-\#warnings>)
```
Comment (mikedld) [inline macosx/CMakeLists.txt:25]:
```suggestion
    $<$<AND:$<NOT:$<STREQUAL:${CMAKE_GENERATOR},Xcode>>,$<COMPILE_LANGUAGE:C,CXX>>:-Wno-\#warnings>)
```

**Comment by Coeur** [comment]
Remaining change is now much smaller. @nevack @mikedld

**Review by nevack — APPROVED** [review]
Note: Update CMake Xcode generator to use new build system.

File: CMakeLists.txt (hunk 1 above)
Comment (ckerr) [reply CMakeLists.txt:6]: @nevack this is still showing as an unresolved conversation. Is any decision / discussion still needed, or can this be resolved?

Comment (nevack) [reply CMakeLists.txt:6]: Sorry, I wasn't aware we need to resolve this somehow.
TBH, I can't even find the control to mark this as resolved now

**Pull request merged**